lib/commands/record-screen.js

```diff
@@ -0,0 +1,339 @@
+    };
+    if (user && pass) {
+      options.auth = {user, pass};
+    }
+  } else if (remoteUrl.protocol.startsWith('ftp')) {
+    options = {
+      host: remoteUrl.hostname,
+      port: remoteUrl.port || 21,
+    };
+    if (user && pass) {
+      options.user = user;
+      options.pass = pass;
+    }
+  }
+  await net.uploadFile(localFile, remotePath, options);
+  return '';
+}
+
+async function requireFfmpegPath () {
+  try {
+    return await fs.which(FFMPEG_BINARY);
+  } catch (e) {
+    log.errorAndThrow(`${FFMPEG_BINARY} has not been found in PATH. ` +
+      `Please make sure it is installed`);
+  }
+}
+
+class ScreenRecorder {
+  constructor (videoPath, opts = {}) {
+    this._videoPath = videoPath;
+    this._process = null;
+    this._fps = (opts.fps && opts.fps > 0) ? opts.fps : DEFAULT_FPS;
+    this._audioInput = opts.audioInput;
+    this._captureCursor = opts.captureCursor;
+    this._captureClicks = opts.captureClicks;
+    this._preset = opts.preset || DEFAULT_PRESET;
+    this._videoFilter = opts.videoFilter;
+    this._timeLimit = (opts.timeLimit && opts.timeLimit > 0)
+      ? opts.timeLimit
+      : DEFAULT_TIME_LIMIT;
+  }
+
+  async getVideoPath () {
+    return (await fs.exists(this._videoPath)) ? this._videoPath : '';
+  }
+
+  isRunning () {
+    return !!(this._process?.isRunning);
+  }
+
+  async _enforceTermination () {
+    if (this._process && this.isRunning()) {
+      log.debug('Force-stopping the currently running video recording');
+      try {
+        await this._process.stop('SIGKILL');
+      } catch (ign) {}
+    }
+    this._process = null;
+    const videoPath = await this.getVideoPath();
+    if (videoPath) {
+      await fs.rimraf(videoPath);
+    }
+    return '';
+  }
+
+  async start () {
+    const ffmpeg = await requireFfmpegPath();
+
+    const args = [
+      '-loglevel', 'error',
+      '-t', `${this._timeLimit}`,
+      '-f', 'gdigrab',
+      ...(this._captureCursor ? ['-capture_cursor', '1'] : []),
+      ...(this._captureClicks ? ['-capture_mouse_clicks', '1'] : []),
+      '-framerate', `${this._fps}`,
+      '-i', 'desktop',
+      ...(this._audioInput ? ['-f', 'dshow', '-i', `audio=${this._audioInput}`] : []),
+      '-vcodec', 'libx264',
+      '-preset', this._preset,
+      '-tune', 'zerolatency',
+      '-pix_fmt', 'yuv420p',
+      '-movflags', '+faststart',
+      '-fflags', 'nobuffer',
+      '-f', DEFAULT_EXT,
+      '-r', `${this._fps}`,
+      ...(this._videoFilter ? ['-filter:v', this._videoFilter] : []),
+    ];
+
+    const fullCmd = [
+      ffmpeg,
+      ...args,
+      this._videoPath,
+    ];
+    this._process = new SubProcess(fullCmd[0], fullCmd.slice(1), {
+      windowsHide: true,
+    });
+    log.debug(`Starting ${FFMPEG_BINARY}: ${util.quote(fullCmd)}`);
+    this._process.on('output', (stdout, stderr) => {
+      if (_.trim(stdout || stderr)) {
+        log.debug(`[${FFMPEG_BINARY}] ${stdout || stderr}`);
```

Comment:
If the output has multiple lines, only the first will have the `[${FFMPEG_BINARY}]` prefix.

Reply:
I would say this is fine. There is no need to see this prefix for each log line